### PR TITLE
Modify README file

### PR DESCRIPTION
**Pull request by jingyimei**
This commit updates the README file from the following perspective:
1. Removed instructions to build gppkg, since it requires make_sync_tool
which is not open source.
2. Removed loading spatial_ref_sys.sql file for enabling geospatial,
since it is not actually used.
3. Add more instructions on setting up raster and gdal driver.
4. Modified some phrases, and reordered some paragraphs.

### CODE REVIEW

**Comment by fmcquillan99** [comment]
Recommended re-wording for clarity:

To compile geospatial from source code, please install the third-party 
libraries as described in README.libs.
For normal use without raster, please install json-c, geos and proj.4.
To enable raster functionality, please install gdal and expat. 
The minimum version requirements are listed in Makefile.version.

File: README.md [hunk 1]
```diff
@@ -1,75 +1,67 @@
...
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/postgis_comments.sql
-psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/spatial_ref_sys.sql
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/rtpostgis.sql
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/raster_comments.sql
 ```
-Besides, to configure raster utilities, please set following variables into env of both
-master and segments, and restart the databases.
+
+Besides, to configure raster utilities, please set following variables into env of both master and segments. A suggested way to do this is to add those variable settings into your `$GPHOME/greenplum_path.sh` file to ensure they get set in all the segments. After setting them, restart the database.
+
```
Comment (njayaram2) [inline README.md]: `After setting them, restart the database` is an important step, it might be good to emphasize it. Not sure how. We can either make it bold, or as a separate sentence by itself, after the three exports below.

File: README.md [hunk 2]
```diff
@@ -1,75 +1,67 @@
...
 ```
 export GDAL_DATA=$GPHOME/share/gdal
 export POSTGIS_ENABLE_OUTDB_RASTERS=0
 export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL
 ```
 
-In near future we plan to move them in GUCs after we backport necessary features
-onto gpdb repo.
+Note: by default, all the gdal drivers are disabled. To enable specific types of gdal driver, please refer to this [postgis manual](http://postgis.net/docs/manual-2.1/postgis_installation.html#install_short_version). An example can be like this:
+
```
Comment (njayaram2) [inline README.md]: I am not certain, but doing this might lead to some failures in `make check`. There are tests which expect a message related to drivers being disabled. This is just an FYI.

File: README.md [hunk 3]
```diff
@@ -1,75 +1,67 @@
...
-## sub-directories under planing
-1. postgis
-  * geometry
-  * raster
-2. trajectory
-3. utilities
-
-## how to compile it
+## How to compile it
 To compile geospatial form source code, please install the follow third-party
```
Comment (kaknikhil) [inline README.md]: /s/follow/following

File: README.md [hunk 4]
```diff
@@ -1,75 +1,67 @@
...
 libraries first by following README.libs.
```
Comment (kaknikhil) [inline README.md]: Can we add a link to README.libs which is in postgis/README.libs

File: README.md [hunk 5]
```diff
@@ -1,75 +1,67 @@
...
 For normal use without raster, please install json-c, geos and proj.4
 To enbale raster function, plese install gdal and expat. The minimum version
```
Comment (kaknikhil) [inline README.md]: /s/enbale/enable

File: README.md [hunk 6]
```diff
@@ -1,75 +1,67 @@
...
 requirments are listed in Makefile.version.
 
 Before setup the geospatial, please make sure the GPDB is installed correctly.
```
Comment (kaknikhil) [inline README.md]: `Before setting up geospatial` instead of `Before setup the geospatial`

File: README.md [hunk 7]
```diff
@@ -1,75 +1,67 @@
...
-To configure the geospatial, please use following command:
+To compile and install geospatial, use following command:
+
 ```
-./configure --with-pgconfig="Your gpdb location"/bin/pg_config --with-raster
---without-topology --prefix=$GPHOME
+./configure --with-pgconfig="Your gpdb location"/bin/pg_config --with-raster --without-topology --prefix=$GPHOME
```
Comment (kaknikhil) [inline README.md]: maybe add a `cd postgis/build/postgis-2.1.5/` before the configure command. There is no configure executable in the root dir

Comment (kaknikhil) [reply README.md]: we should also mention that if psql is in path, then there is no need to use --use-pgconfig

File: README.md [hunk 8]
```diff
@@ -1,75 +1,67 @@
 # geospatial repo
+PostGIS 2.1.5 for GreenPlum 5.x+
 
-## licence
+## Licence
```
Comment (kaknikhil) [inline README.md]: /s/Licence/License

File: README.md [hunk 9]
```diff
@@ -1,75 +1,67 @@
...
+./configure --with-pgconfig="Your gpdb location"/bin/pg_config --with-raster --without-topology --prefix=$GPHOME
+make
+make install	
 ```
 
-If configuration is successfully, then please run _make_ to compile the geospatial
-and run make install to install the geospatial to the GPDB. If you build from
-the extended PostGIS-2.x directory, you may compile with following command:
+If you build from
+the extended PostGIS-2.x directory, you may compile with the following command:
```
Comment (kaknikhil) [inline README.md]: `the extended PostGIS-2.x directory` , i am assuming this refers to `postgis/build/postgis-2.1.5` . If yes, is there any other directory you can build from ?

File: README.md [hunk 10]
```diff
@@ -1,75 +1,67 @@
...
 
-## how to distribute it
-The geospatial has bulit-in function to build the geospatial as a gppkg to allow
-user to install the geospatial directly into GPDB without compiling.
+Here USE_PGXS will specify the correct install path to gpdb.
 
-To build the gppkg, please make sure the source code of GPDB is downloaded and
-```make sync_tools``` is run correctly.
+## How to use it
+After installing geospatial extention, run the following commands to enable it:
```
Comment (kaknikhil) [inline README.md]: /s/extention/extension

File: README.md [hunk 11]
```diff
@@ -1,75 +1,67 @@
...
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/postgis.sql
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/postgis_comments.sql
-psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/spatial_ref_sys.sql
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/rtpostgis.sql
 psql -d mydatabase -f ${GPHOME}/share/postgresql/contrib/postgis-2.1/raster_comments.sql
 ```
-Besides, to configure raster utilities, please set following variables into env of both
-master and segments, and restart the databases.
+
+Besides, to configure raster utilities, please set following variables into env of both master and segments. A suggested way to do this is to add those variable settings into your `$GPHOME/greenplum_path.sh` file to ensure they get set in all the segments. After setting them, restart the database.
```
Comment (kaknikhil) [inline README.md]: maybe reword it something like this ` To configure raster utilities, please set the following environment variables on the master host and all the segment hosts. A suggested way to do this is to add these variables  into your `$GPHOME/greenplum_path.sh` file to ensure they get set in all the segment hosts and the master. After setting them, restart the database.`

File: README.md [hunk 12]
```diff
@@ -1,75 +1,67 @@
...
+
 ```
 export GDAL_DATA=$GPHOME/share/gdal
 export POSTGIS_ENABLE_OUTDB_RASTERS=0
 export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL
 ```
 
-In near future we plan to move them in GUCs after we backport necessary features
-onto gpdb repo.
+Note: by default, all the gdal drivers are disabled. To enable specific types of gdal driver, please refer to this [postgis manual](http://postgis.net/docs/manual-2.1/postgis_installation.html#install_short_version). An example can be like this:
```
Comment (kaknikhil) [inline README.md]: /s/driver/drivers

File: README.md [hunk 13]
```diff
@@ -1,75 +1,67 @@
...
+
+```
+POSTGIS_GDAL_ENABLED_DRIVERS="GTiff PNG JPEG GIF XYZ"
+```
+
+In near future we plan to move them in GUCs after we backport necessary features onto gpdb repo.
```
Comment (kaknikhil) [inline README.md]: maybe say `we plan to create GUCs for these variables after we backport necessary features into the gpdb repository.`

File: README.md [hunk 14]
```diff
@@ -1,75 +1,67 @@
...
+
+## Sub-directories under planing
+1. postgis
+  * geometry
```
Comment (kaknikhil) [inline README.md]: this markdown syntax isn't indenting the `geometry` and `raster` sub sections

File: README.md [hunk 15]
```diff
@@ -1,75 +1,67 @@
...
+  * raster
+2. trajectory
+3. utilities
 
-## last update date
-	Kuien Liu, Haozhou Wawng, 26 May 2016
+## Last update date
```
Comment (kaknikhil) [inline README.md]: maybe add a new line after each update

File: README.md (hunk 2 above)
Comment (jingyimei) [reply README.md]: I understand this, the point I want to make here is for later use. To pass all the install check, we do need to disable all the drivers. Let me try to rephrase this.

File: README.md (hunk 14 above)
Comment (jingyimei) [reply README.md]: I did see indentation in my macdown. What do you mean here?

File: README.md (hunk 1 above)
Comment (jingyimei) [reply README.md]: rephrased

File: README.md (hunk 9 above)
Comment (jingyimei) [reply README.md]: Yes it refers to `postgis/build/postgis-2.1.5`. The other way is to build gppkg from the package folder. Since we already removed the instructions for building gppkg, we can remove this explanation too.

File: README.md [hunk 16]
```diff
@@ -1,75 +1,61 @@
 # geospatial repo
+PostGIS 2.1.5 for GreenPlum 5.x+
 
-## licence
+## License
+This project is developed under GPL v2, because PostGIS is GPL v2.
 
-This project is developing under GPL v2, because PostGIS is GPL v2.
+## How to compile it
+Currently, we support building geospatial on rhel/centos. To compile geospatial form source code, please install the following third-party libraries as described in [README.libs](https://github.com/greenplum-db/geospatial/blob/master/postgis/README.libs).
```
Comment (njayaram2) [inline README.md]: `README.libs` does not have the installation instructions for `gdal` and `expat`. I think it is useful to have a verbose README that details those steps too. Without installing gdal (and expat?), we cannot run the `./configure` command that follows in this doc.

Comment (jingyimei) [reply README.md]: make sense.

File: README.md [hunk 17]
```diff
@@ -1,75 +1,61 @@
...
 
-If configuration is successfully, then please run _make_ to compile the geospatial
-and run make install to install the geospatial to the GPDB. If you build from
-the extended PostGIS-2.x directory, you may compile with following command:
-```
-make USE_PGXS=1 clean all install
-```
 Here USE_PGXS will specify the correct install path to gpdb.
 
+Note that if psql is in path, there is no need to use --use-pgconfig flag.
```
Comment (njayaram2) [inline README.md]: `--use-pgconfig` or `--with-pgconfig`?

File: postgis/README.libs [hunk 18]
```diff
@@ -1,32 +1,23 @@
...
 3. sudo yum install -y proj-devel
 
-Installed:
-  proj-devel.x86_64 0:4.8.0-4.el7
-
-  Dependency Installed:
-    proj.x86_64 0:4.8.0-4.el7
-
-4. gdal
+4. sudo yum install -y expact
```
[the remark this answers is not in the document]
Comment (jingyimei) [reply postgis/README.libs]: Sorry for the typo. Will correct that.

File: README.md (hunk 17 above)
Comment (jingyimei) [reply README.md]: This one is from the original readme, and I believe it should be `--with-pgconfig`. will correct it.

File: README.md [hunk 19]
```diff
@@ -1,75 +1,61 @@
...
+Before setting up geospatial, please make sure GPDB is installed correctly.
+To compile and install geospatial, use following command:
 
-Before setup the geospatial, please make sure the GPDB is installed correctly.
-To configure the geospatial, please use following command:
 ```
-./configure --with-pgconfig="Your gpdb location"/bin/pg_config --with-raster
---without-topology --prefix=$GPHOME
+cd postgis/build/postgis-2.1.5/
+./configure --with-pgconfig=$GPHOME/bin/pg_config --with-raster --without-topology --prefix=$GPHOME
```
Comment (jasminepoly) [inline README.md]: When I try  to run this command I get this error :
checking PostgreSQL version... PostgreSQL 8.3.23
configure: error: PostGIS requires PostgreSQL >= 9.0
Is this the right version of PostGIS?
